src/Form43Checker.py

```diff
@@ -89,32 +89,19 @@ def __getForm(self, file):
         else:
             return "invalid"
 
+    def __listInvalidFiles(self):
+        for file in self.__organizer.findFiles("", "*.es[pm]"):
+            if self.__testFile(file):
+                yield file
+
     def __scanPlugins(self):
         if self.__organizer.managedGame().gameName() != "Skyrim Special Edition":
             return False
 
-        def isPluginFile(name):
-            extension = Path(name).suffix
-            extension = extension.lower()
-            return extension == ".esp" or extension == ".esm"
-
-        fileList = self.__organizer.findFiles("", isPluginFile)
-
-        for file in fileList:
-            if self.__testFile(file):
-                return True
-
-        return False
+        return any(True for _ in self.__listInvalidFiles())
```

Comment:
This seems like it's unnecessarily verbose and probably slow. An empty iterator is already implicitly convertible to `bool`, so you could just `return self.__listInvalidFiles()`, and if you want to be explicit, there are more clear ways to check if an iterator is empty than converting it to another iterator that contains `True` for every element in the original and checking if that iterator contains `any` `True`s.

Reply:
That's not correct. `bool()` on a generator object will always returns `True`, even if the generator is empty, which is not what I want here. The other way is to do something like:

```python
for file in self.__listInvalidFiles():
  return True
return False
```

Which is not really better than the actual version. `any(self.__listInvalidFiles())` would probably work here since there is no way an empty file can be return by `self.__listInvalidFiles()` but I’m not fan of it.

Reply:
Huh. That's pretty terrible. Most other things like that *are* convertible to `bool`, but apparently they took the *once it's generated it's not held in memory* idea to an extreme and didn't even provide a `peek` or `has_next`.

Either way, I still think the existing version is less good than any of the others. The options are variations on:

* ```py
  return any(True for _ in self.__listInvalidFiles())
  ```
  which I don't like as it makes it look like it was written by someone who didn't know what they were doing, rather than by someone trying to work around a language flaw. Even with a comment explaining it, it still looks like there's no way it can be the simplest option.

* ```py
  for file in self.__listInvalidFiles():
      return True
  return false
  ```
  which I'd regard as okay-ish.

* ```py
  any(self.__listInvalidFiles())
  ```
  which is kind of cheating, but is immediately obvious what it's doing and is nice and concise, so is probably my favourite so far.

* ```py
  try:
      next(self.__listInvalidFiles())
      return True
  except StopIteration:
      return False
  ```
  which I believe is supposed to be the most Pythonic way (as exceptions aren't really any more expensive than any other form of control flow in Python and are intended to be used for things like this, even when the exception happening is the good case). It's quite bulky and won't look clean to anyone who's aware of best practice in practically any other language, though.

* ```py
  return next(self.__listInvalidFiles(), False)
  ```
  which is pretty concise and obvious and isn't cheating, so is probably my overall favourite. It can be tweaked to make it more explicit if that's needed.

I don't think any of the other options I've seen are worth discussing.

Reply:
Well, it's your plugin in the end so... 🤷‍♂️